### ISSUE TITLE
Filter out type import dependencies

### ISSUE THREAD

**Issue by tom2drum** [dependents/node-dependency-tree#172]
Can this tool be configured to exclude all imports of any types (via `import type`) from being considered as file dependencies?

**Comment by tom2drum** [comment]
No, that doesn't work for me. I think this setting only changes the resolver path from `.js` files to `.d.ts` files.

**Comment by boredland** [comment]
ah, this got me too. I don't care about `import type X` and `import { type X }`. Is there a way to not follow those?

**Comment by XhmikosR** [comment]
Untested, wouldn't something like this work?

```js
dependencyTree({
  filename: 'src/foo.ts',
  directory: '.',
  detective: {
    ts: {
      skipTypeImports: true
    }
  }
});
```

**Comment by tom2drum** [comment]
It could, but I don't see such configuration for typescript in the `precinct` package.

**Comment by XhmikosR** [comment]
Please try it and see if it works.

**Comment by XhmikosR** [comment]
https://github.com/dependents/node-dependency-tree/commit/6bec9f01cd40630efb7e82de305c420cbaf14b00#diff-ff073bce3bc0b3dd690ba4dc71bcb75cc2f4cd682e944705db8e21973aad6627R271-R279
